docs/QSG/qemu-vms.md

```diff
@@ -0,0 +1,284 @@
+        --dhcp-leasefile=/var/run/qemu-dnsmasq-$BRIDGE.leases \
+        --dhcp-no-override \
+        ${TFTPROOT:+"--enable-tftp"} \
+        ${TFTPROOT:+"--tftp-root=$TFTPROOT"} \
+        ${BOOTP:+"--dhcp-boot=$BOOTP"}
+    }
+    
+    setup_bridge_nat() {
+        if check_bridge "$1" ; then
+        create_bridge "$1"
+        enable_ip_forward
+        add_filter_rules "$1"
+        start_dnsmasq "$1"
+        fi
+    }
+    
+    setup_bridge_vlan() {
+        if check_bridge "$1" ; then
+        create_bridge "$1"
+        start_dnsmasq "$1"
+        fi
+    }
+    
+    setup_bridge_nat "$BRIDGE"
+    
+    if test "$1" ; then
+        do_ifconfig "$1" 0.0.0.0 up
+        do_brctl addif "$BRIDGE" "$1"
+    fi
+```
+Then add a sudo rule for qemu-system-x86<sub>64</sub> in *etc/sudoers.d*.
+```
+    <username> blackbox=(root) NOPASSWD: /usr/bin/qemu-system-x86_64
+```
+3.  Install guest operating systems.
+
+Run these two comands. Select "boot from cdrom". Follow the installation guide to install the OSes.
+```
+    sudo qemu-system-x86_64 -M q35,accel=kvm,kernel-irqchip=split -cpu host -smp 3 -m 12288 -device intel-iommu,intremap=on -drive file=<image-dir>/daos-server.qcow2,if=virtio -device virtio-net,netdev=mynet0,mac=52:54:00:12:34:56 -netdev tap,id=mynet0 -drive file=<image-dir>/qemu-nvm-disk1.qcow2,if=none,id=nvm1 -device nvme,serial=deadbeef,drive=nvm1 -cdrom <iso-dir>/Rocky-8.9-x86_64-minimal.iso -boot menu=on &
+    sudo qemu-system-x86_64  -M q35,accel=kvm,kernel-irqchip=split -cpu host -smp 1 -m 2048 -device intel-iommu,intremap=on -drive file=<image-dir>/daos-client.qcow2,if=virtio -device virtio-net,netdev=mynet1,mac=52:54:00:12:34:57 -netdev tap,id=mynet1 -cdrom <iso-dir>/Rocky-8.9-x86_64-minimal.iso -boot menu=on
+```
+4.  Install DAOS software.
+
+After OS installation is finished. Remove the `-cdrom` and `-boot` options from the QEMU commands to boot VMs normally. Next is to install DAOS software.
+```
+    sudo qemu-system-x86_64 -M q35,accel=kvm,kernel-irqchip=split -cpu host -smp 3 -m 12288 -device intel-iommu,intremap=on -drive file=<image-dir>/daos-server.qcow2,if=virtio -device virtio-net,netdev=mynet0,mac=52:54:00:12:34:56 -netdev tap,id=mynet0 -drive file=<image-dir>/qemu-nvm-disk1.qcow2,if=none,id=nvm1 -device nvme,serial=deadbeef,drive=nvm1 &
+    sudo qemu-system-x86_64  -M q35,accel=kvm,kernel-irqchip=split -cpu host -smp 1 -m 2048 -device intel-iommu,intremap=on -drive file=<image-dir>/daos-client.qcow2,if=virtio -device virtio-net,netdev=mynet1,mac=52:54:00:12:34:57 -netdev tap,id=mynet1 
+```
+I follow these [steps](https://docs.daos.io/v2.4/QSG/setup_rhel/#start-the-daos-agents) to install both the DAOS server, DAOS admin, and DAOS client. I install daos-server on the first VM, and install both daos-admin and daos-client on the second VM. If you have issues with Ceph Pacific Repo you can edit `/etc/yum.repos.d/CentOS-Ceph-Pacific.repo` by commenting out mirrorlist and changing baseurl as below.
```

Comment:
From what I know, the best practice is to not install the daos-admin (i.e. dmg command) on a DAOS client node, but to install it on one of the DAOS servers or on dedicated node.

Reply:
Is it related to security concern? I can relocate daos-admin to daos-server.

Reply:
Yes, from what I know: dmg command is the sysadmin command which should not be run by the DAOS client user.  The dmg command is indeed aimed for managing the DAOS server: telemetry, storage, pool creation, etc.  On the other side, the daos command is aimed for the DAOS client user for managing daos container and other stuff.
Thus, it should be a good thing to relocate daos-admin to daos-server.